plugins/CoreHome/angularjs/common/services/piwik.js

```diff
@@ -68,10 +68,11 @@
 
             // Cache server-rendered page title
             originalTitle = originalTitle || document.title;
-            var titleParts = originalTitle.split('-');
-            var dateString = ' ' + piwikPeriods.parse(period, date).getPrettyString() + ' ';
-            titleParts.splice(1, 0, dateString);
-            document.title = titleParts.join('-');
+
+            if (0 === originalTitle.indexOf(piwik.siteName)) {
+                var dateString = ' - ' + piwikPeriods.parse(period, date).getPrettyString() + ' ';
+                document.title = piwik.siteName + dateString + originalTitle.substr(piwik.siteName.length);
+            }
```

Comment:
Are there any cases where the date should be added and the document title does not include the site name? If there are, then this won't add the dates to those titles.

Reply:
Wasn't 100% sure if that might be the case somewhere. That's the reason for the if.
But the target was to put the date after the website name again. See https://github.com/matomo-org/matomo/pull/12222/files#diff-1eb3ce37602c6f5a47792ccdf0bce2ff

Reply:
your solution solves my problem, thank you. see #13696